module_utils/certificate/providers/certmonger.py

```diff
@@ -110,6 +110,7 @@ def exists_in_certmonger(self):
 
     def request_certificate(self):
         """Issue or update a certificate using certmonger."""
+        # pylint: disable=useless-else-on-loop
```

Comment:
Why is the `else` useless?  Does the linter think that e.g. `self.csr.dns` will always have at least 1 element?

Reply:
`pylint` complains that there is an `else` in a `for` loop when there is no `break` inside the loop. The `for-else` was designed for fallback in cases where the programmer is looking for something in the loop and can't find it.
Since we are using it in a different way we need to disable this warning.

Initially I've used `if not` to avoid the lint complains without having to explicitly silent it.

Reply:
Ok - I see - `for/else` is supposed to be used with a `break` - I have typically used it without a break, but not in linted code . . . then I'll leave it up to you if you want to revert back to the `if not xxx` version, or keep the current code with the pylint suppression.

Reply:
Ok. I fine with the current way.